Tests/ApolloTests/Interceptors/MultipartResponseParsingInterceptorTests.swift

```diff
@@ -0,0 +1,355 @@
+import XCTest
+import Nimble
+@testable import Apollo
+import ApolloAPI
+import ApolloInternalTestHelpers
+
+final class MultipartResponseParsingInterceptorTests: XCTestCase {
+
+  private class ErrorRequestChain: RequestChain {
+    var isCancelled: Bool = false
+    var error: Error? = nil
+
+    init() {}
+
+    func kickoff<Operation>(
+      request: Apollo.HTTPRequest<Operation>,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {}
+
+    func proceedAsync<Operation>(
+      request: Apollo.HTTPRequest<Operation>,
+      response: Apollo.HTTPResponse<Operation>?,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {}
+
+    func cancel() {}
+
+    func retry<Operation>(
+      request: Apollo.HTTPRequest<Operation>,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {}
+
+    func handleErrorAsync<Operation>(
+      _ error: Error,
+      request: Apollo.HTTPRequest<Operation>,
+      response: Apollo.HTTPResponse<Operation>?,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {
+      self.error = error
+    }
+
+    func returnValueAsync<Operation>(
+      for request: Apollo.HTTPRequest<Operation>,
+      value: Apollo.GraphQLResult<Operation.Data>,
+      completion: @escaping (Result<Apollo.GraphQLResult<Operation.Data>, Error>) -> Void
+    ) {}
+  }
+
+  // MARK: - Error tests
```

Comment:
These error tests are done directly on the interceptor because in normal operation there are other safeguards in earlier interceptors that _should_ never make these possible but with optionals we need to handle unintended behaviour. Doing this on only the interceptor ensures we can test all error cases.

Reply:
So are the safeguards in earlier interceptors ignored for multipart requests?